diff --git a/test/blockchain/NodeTest.java b/test/blockchain/NodeTest.java
index c7e7b1f..1a271e8 100644
--- a/test/blockchain/NodeTest.java
+++ b/test/blockchain/NodeTest.java
@@ -41,12 +41,12 @@ abstract class NodeTest extends Test
 
     protected static final int BROADCAST_TIMEOUT_MS = 5000;
     protected static final int SLEEP_TIMEOUT = 30;
-    protected static final int CLIENT_TIMEOUT = 10;
+    protected static final int CLIENT_TIMEOUT = 30;
 
     protected static final int KEYCHAIN_ID = 1;
     protected static final int VOTECHAIN_ID = 2;
     protected static final int[] CHAIN_IDS = {KEYCHAIN_ID, VOTECHAIN_ID};
-    protected static final String[] CHAIN_PROOFS = {"00000", ""};
+    protected static String[] CHAIN_PROOFS;
     protected static final String[] BROADCAST_TYPES = {"PRECOMMIT", "COMMIT"};
 
     private List<Process> servers = new ArrayList<>();
@@ -56,6 +56,22 @@ abstract class NodeTest extends Test
     abstract protected void perform() throws TestFailed;
 
 
+    /**
+     * Constructor of the test
+     */
+    protected NodeTest()
+    {
+        if (Config.node_config.startsWith("python"))
+        {
+            CHAIN_PROOFS = new String[]{"000", ""};
+        }
+        else
+        {
+            CHAIN_PROOFS = new String[]{"00000", ""};
+        }
+    }
+
+
     /**
      *      Spawns processes that start blockchain nodes.
      *
